aws/rust-runtime/aws-config/src/imds/client/token.rs

```diff
@@ -47,158 +51,226 @@ const TOKEN_REFRESH_BUFFER: Duration = Duration::from_secs(120);
 
 const X_AWS_EC2_METADATA_TOKEN_TTL_SECONDS: &str = "x-aws-ec2-metadata-token-ttl-seconds";
 const X_AWS_EC2_METADATA_TOKEN: &str = "x-aws-ec2-metadata-token";
+const IMDS_TOKEN_AUTH_SCHEME: AuthSchemeId = AuthSchemeId::new(X_AWS_EC2_METADATA_TOKEN);
 
 /// IMDS Token
 #[derive(Clone)]
 struct Token {
     value: HeaderValue,
     expiry: SystemTime,
 }
+impl fmt::Debug for Token {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("Token")
+            .field("value", &"** redacted **")
+            .field("expiry", &self.expiry)
+            .finish()
+    }
+}
 
-/// Token Middleware
-///
-/// Token middleware will load/cache a token when required and handle caching/expiry.
+/// Token Runtime Plugin
 ///
-/// It will attach the token to the incoming request on the `x-aws-ec2-metadata-token` header.
-#[derive(Clone)]
-pub(super) struct TokenMiddleware {
-    client: Arc<aws_smithy_client::Client<DynConnector, MapRequestLayer<UserAgentStage>>>,
-    token_parser: GetTokenResponseHandler,
-    token: ExpiringCache<Token, ImdsError>,
-    time_source: SharedTimeSource,
-    endpoint: Uri,
-    token_ttl: Duration,
+/// This runtime plugin wires up the necessary components to load/cache a token
+/// when required and handle caching/expiry. This token will get attached to the
+/// request to IMDS on the `x-aws-ec2-metadata-token` header.
+#[derive(Debug)]
+pub(super) struct TokenRuntimePlugin {
+    components: RuntimeComponentsBuilder,
+}
+
+impl TokenRuntimePlugin {
+    pub(super) fn new(
+        common_plugin: SharedRuntimePlugin,
+        time_source: SharedTimeSource,
+        token_ttl: Duration,
+    ) -> Self {
+        Self {
+            components: RuntimeComponentsBuilder::new("TokenRuntimePlugin")
+                .with_auth_scheme(SharedAuthScheme::new(TokenAuthScheme::new()))
```

Comment:
```suggestion
                .with_auth_scheme(TokenAuthScheme::new())
```

Reply:
Fixed in 1d68fdfda697e1e50c83951acfef78bebdcdc6d3